packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/zoomlevel/CameraZoom.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera.features.zoomlevel;
+
+import android.graphics.Rect;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.math.MathUtils;
+
+public final class CameraZoom {
+  public static final float DEFAULT_ZOOM_FACTOR = 1.0f;
+
+  @NonNull private final Rect cropRegion = new Rect();
+  @Nullable private final Rect sensorSize;
+
+  public final float maxZoom;
+  public final boolean hasSupport;
+
+  public static CameraZoom create(@Nullable final Rect sensorArraySize, final Float maxZoom) {
+    return new CameraZoom(sensorArraySize, maxZoom);
+  }
+
+  private CameraZoom(@Nullable final Rect sensorArraySize, final Float maxZoom) {
+    this.sensorSize = sensorArraySize;
+
+    if (this.sensorSize == null) {
+      this.maxZoom = DEFAULT_ZOOM_FACTOR;
+      this.hasSupport = false;
+      return;
+    }
+
+    this.maxZoom =
+        ((maxZoom == null) || (maxZoom < DEFAULT_ZOOM_FACTOR)) ? DEFAULT_ZOOM_FACTOR : maxZoom;
+
+    this.hasSupport = (Float.compare(this.maxZoom, DEFAULT_ZOOM_FACTOR) > 0);
+  }
+
+  public Rect computeZoom(final float zoom) {
+    if (sensorSize == null || !this.hasSupport) {
+      return null;
+    }
+
+    final float newZoom = MathUtils.clamp(zoom, DEFAULT_ZOOM_FACTOR, this.maxZoom);
```

Comment:
Why is the default a minimum?

Reply:
The constant `DEFAULT_ZOOM_FACTOR` is named incorrect and should have been named `MIN_ZOOM_LEVEL`.